lib/turn.rb

```diff
@@ -1,6 +1,40 @@
 class Turn
-  attr_reader
-  
-  def initialize
+  attr_reader :message, :codemaker, :codebreaker
+
+  def initialize(codemaker, codebreaker, message)
+    @codemaker = codemaker
+    @codebreaker = codebreaker
+    @message = message
+    @guess = nil
+    @winner = false
   end
-end
\ No newline at end of file
+
+  def find_code
+    @codemaker.code
+  end
+
+  def guess_length_check
+    if @codebreaker.guess.length < 4
+        @message.short_answer
+    elsif @codebreaker.guess.length > 4
+           @message.long_answer
+    end
+  end
+
+  def has_won?
+    @winner = true if @codebreaker.guess == @codemaker.code
+  end
+
+  def num_correct_colors
```

Comment:
The indentation formatting for this one is a little wonky, we should address this in out next push.